### PR TITLE
build logic fixes

### LINKED ISSUE #1356: make clean broken (was: New rm + rd clean is not quiet)
On OSX, make clean in drake results in:
```
tk-sloth:drake twan$ make clean
-- CMAKE_INSTALL_PREFIX=/Users/twan/code/drake/build
-- Preparing to build cmake with dependencies: 
-- Preparing to build eigen with dependencies: 
-- Preparing to build lcm with dependencies: 
-- Preparing to build libbot with dependencies: lcm
-- Preparing to build bullet with dependencies: 
-- Preparing to build spotless with dependencies: 
-- Preparing to build director with dependencies: lcm;libbot
-- Preparing to build drake with dependencies: cmake;eigen;lcm;libbot;bullet
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/twan/code/drake/pod-build
Scanning dependencies of target clean-libbot

-------------------------------------------
-- bot2-core
-------------------------------------------

-------------------------------------------
-- bot2-vis
-------------------------------------------

-------------------------------------------
-- bot2-procman
-------------------------------------------

-------------------------------------------
-- bot2-lcmgl
-------------------------------------------

-------------------------------------------
-- bot2-lcm-utils
-------------------------------------------

-------------------------------------------
-- bot2-param
-------------------------------------------

-------------------------------------------
-- bot2-frames
-------------------------------------------
Built target clean-libbot
Scanning dependencies of target clean-bullet
make[5]: rd: No such file or directory
make[5]: rd: No such file or directory
Built target clean-bullet
Error: pod-build is not a directory
make[5]: *** [clean] Error 1
make[4]: *** [CMakeFiles/clean-spotless] Error 2
make[3]: *** [CMakeFiles/clean-spotless.dir/all] Error 2
make[2]: *** [CMakeFiles/clean-all.dir/rule] Error 2
make[1]: *** [clean-all] Error 2
make: *** [clean] Error 2
```

Can we move the removal of build and pod-build into the CMakeLists.txt and use e.g. additional_make_clean_files or just file(REMOVE [file1 ...]) instead?

### PR DESCRIPTION
**Pull request by RussTedrake**
resolves #1356
also resolves the WITH_ALL_SUPPORTED_EXTERNALS logic error reported in #1361

### CODE REVIEW

**Comment by tkoolen** [comment]
You probably figured it out already; it looks like bertini, gloptipoly, sedumi and snopt are set up to use ssh, but the build servers probably aren't.

File: drake/Makefile [hunk 1]
```diff
@@ -33,13 +33,9 @@ endif
...
 .PHONY: clean
 clean:
-	cmake --build pod-build --target clean
-# for windows cmd shell
-	@-rd /s pod-build        
-	@-rd /s build
-# for everyone else
-	@-rm -rf pod-build
-	@-rm -rf build
+	cmake --build pod-build --target clean-all
```
Comment (tkoolen) [inline drake/Makefile]: Maybe I don't understand, but I thought the target should be clean, not clean-all. I don't think the CMakeLists.txt in drake-distro/drake defines a clean-all target. Same for all of the externals.

Comment (RussTedrake) [inline drake/Makefile]: absolutely right.  drake-distro/Makefile needs clean-all, but the rest just clean (a copy and paste bug).  sigh.  i'll fix it.  good catch!

**Comment by RussTedrake** [comment]
been working on it now

**Pull request merged**